### ISSUE TITLE
Windows systems think they are on Gitpod

### ISSUE THREAD

**Issue by bkimminich** [dword-design/is-gitpod#94]
The current `isGitpod()` check triggers positive on Windows 10/11 machines. This of course leads to unwanted side effects, for example https://github.com/juice-shop/juice-shop/issues/1876.

**Comment by dword-design** [comment]
Thanks for reporting. The lib detects GitPod simply by checking for the existing of the `gp` command line tool. Do you know which tool is installed on windows that has the same name?

**Comment by bkimminich** [comment]
I'll check later what the `gp.exe|com|bat` tool is that I seem to have on my machine... 😅

**Comment by bkimminich** [comment]
I do not have anything `gp` on my machine...
![image](https://user-images.githubusercontent.com/3531020/188511102-fb280888-4eae-4566-a7bc-a16584b98145.png)

**Comment by bkimminich** [comment]
It seems the error message is not what you expect it to be...
![image](https://user-images.githubusercontent.com/3531020/188511497-1dbfa098-0136-4fb4-bc2a-5e15af1c89fe.png)

**Comment by dword-design** [comment]
Does `which gp` return an error code? When you are in a Linux subsystem you can run `echo $?` I guess.

**Comment by bkimminich** [comment]
https://github.com/juice-shop/juice-shop/issues/1876#issuecomment-1237807157

It seems this is not just a Windows problem. Has been reproduced on Kali Linux as well.

Maybe `which` has changed their returned message text when a command hasn't been found?

**Comment by bkimminich** [comment]
> echo $?

```
bjoer_000@MSI_GT70_BKI MINGW64 /c/Data/GitHub/juice-shop (develop)
$ npx which gp

bjoer_000@MSI_GT70_BKI MINGW64 /c/Data/GitHub/juice-shop (develop)
$ echo $?
1

bjoer_000@MSI_GT70_BKI MINGW64 /c/Data/GitHub/juice-shop (develop)
$ echo $?
0
```

Is this what you mean?

**Comment by bkimminich** [comment]
```
bjoer_000@MSI_GT70_BKI MINGW64 /c/Data/GitHub/juice-shop (develop)
$ npx which node
C:\Program Files\nodejs\node.EXE

bjoer_000@MSI_GT70_BKI MINGW64 /c/Data/GitHub/juice-shop (develop)
$ echo $?
0

bjoer_000@MSI_GT70_BKI MINGW64 /c/Data/GitHub/juice-shop (develop)
$ npx which gp

bjoer_000@MSI_GT70_BKI MINGW64 /c/Data/GitHub/juice-shop (develop)
$ echo $?
1
```

**Comment by dword-design** [comment]
@bkimminich Yes that's it. It returns 1, so it returns an error. It should then run into the catch block and either return false or rethrow the error, but not true. Can you do the same with `is-gitpod` command-line tool, so `npx is-gitpod`?
